.github/workflows/deploy.yml

```
@@ -0,0 +1,62 @@
+name: GitHub Pages Deploy
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: pages
+  cancel-in-progress: false
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - uses: actions/setup-node@v4
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build Slides > 2025 > Build Tooling
+        working-directory: 2025/build-tooling
+        run: |
+          npx slidev build --base /${{github.event.repository.name}}/2025/build-tooling/ --out ../../dist/2025/build-tooling
+
+      - name: Build Slides > 2025 > Using Components 2
+        working-directory: 2025/using-components-2
+        run: |
+          npx slidev build --base /${{github.event.repository.name}}/2025/using-components-2/ --out ../../dist/2025/using-components-2/
+
+      - name: Collect build outputs
+        run: |
+          mv -f 2024 dist/2024
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v4
```

Small nitpick that we could use v5 https://github.com/actions/configure-pages/releases

```suggestion
        uses: actions/configure-pages@v5
```